ray-operator/controllers/ray/expectations/scale_expectations_test.go

```diff
@@ -0,0 +1,142 @@
+package expectations
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientFake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestRayClusterExpectationsHeadPod(t *testing.T) {
```

Comment:
What is the expected behavior of the following sequence?

- Expect create head pod
- Expect delete head pod
- IsSatisfied = ?
- Create head pod
- IsSatisfied = ?
- Delete head pod
- IsSatisfied = ?

I ask this because Kruise has a test about this case.

https://github.com/openkruise/kruise/blob/5ce62c948a6bef9cb079a10fb40e9045b5d74e09/pkg/util/expectations/scale_expectations_test.go#L14-L35

Reply:
Our expected behavior
- Expect create head pod
- Expect delete head pod  _[Override the expectation for the same Pod]_
- IsSatisfied = `true`        _[There is no pod in the informer. Satisfied. And delete expectation.]_
- Create head pod
- IsSatisfied = `true` _[No expectation]_
- Delete head pod
- IsSatisfied = `true` _[No expectation]_

This is an interesting test. In our design, the latest desired state of the same pod will override the previous one, meaning we only retain our final expected state. However, this order does not occur during reconciliation. In reality, we expect a pod to be deleted only if it currently exists in the informer.   

In practice, the execution sequence would be:
- reconcile 1
   - Create head pod _[request ApiServer]_
   - Expect create head pod
- reconcile 2
   - if IsSatisfied (create),  Delete head pod;
- reconcile 3
   - if IsSatisfied (delete),  do sth.;